### ISSUE TITLE
Memory leak in q3huff.Reader()

### ISSUE THREAD

**Issue by JKornev** [jkent/python-q3huff#1]
Hi man,

Thanks for the project, I use it in few of my projects and I noticed that a memory in my application constantly grow up and after an investigation I found that an issue located in q3huff.Reader().

To prove it I executed the following script
```
def leak():
    while True:
        writer = q3huff.Writer()
        writer.oob = True
        for i in range(200):
            writer.write_string("hi there hi there hi there hi there hi there ")
        reader = q3huff.Reader(writer.data)
        reader.oob = True
        for i in range(200):
            s = reader.read_string()
```

The leak() function starts dramatically utilize a memory, this is how it looks like after few minutes:
![image](https://user-images.githubusercontent.com/5193564/175789830-85b03cc0-b24a-4abd-ada9-cc82a9a5fd34.png)

Inside a process I see that occupied memory located on a common heap and if we dump few random heap chunks we can see the following content:
![image](https://user-images.githubusercontent.com/5193564/175790018-b24943ac-f430-40b1-bfaf-4da0a4c298ec.png)

So seems like it's definitely Reader or Writer issue but when I commented reader part in the leak() function a leak disappeared, so it's a reader issue.

I do think that leak happens because of an extra reference :
https://github.com/jkent/python-q3huff/blob/73117d4845fc5c4b5fec41df6835b3d595dc3d3c/src/hufflib.c#L546

Suspect that when you call Py_INCREF(result) you add an additional reference that is not needed because 'result' object already referenced.

Regards,
JK

**Comment by jkent** [comment]
Sorry this has fallen through the cracks.  The project this was intended for was abandoned.  I was not very familiar with the python C API when I wrote this initially.  I'll merge your PR and get the module updated on PyPI.

**Comment by jkent** [comment]
Closed with merge of 7d88c6e.
